preserve custom meta in placeholders

avikchaudhuri opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149661

Fixes #147338

Differential Revision: [D71573533](https://our.internmc.facebook.com/intern/diff/D71573533/)

avikchaudhuri committed:
preserve custom meta in placeholders

Fixes #147338

Differential Revision: [D71573533](https://our.internmc.facebook.com/intern/diff/D71573533/)

[ghstack-poisoned]

angelayi commented on torch/_export/utils.py:
```diff
@@ -918,6 +924,8 @@ def _extract_pytree_key(x):
             assert node.name in name_map
             node.name = node.target = name_map[node.name]
+            if node.name in custom_meta:
+                node.meta["custom"] = custom_meta[node.name]
```
It seems a little weird that the "placeholder_naming_pass" is also copying over the metadata.. Is there a better place to add the logic? Maybe there isn't -- originally when I got stuck I was trying to make tracing natively propagate the metadata.

avikchaudhuri replied:
Yeah I gave up on propagating upstream, was getting too deep.

This pass already touches up metadata for custom obj arguments. It also builds a mapping between args, so I thought it would be a good place to put this logic.

avikchaudhuri committed:
Update on "preserve custom meta in placeholders"


Fixes #147338

Differential Revision: [D71573533](https://our.internmc.facebook.com/intern/diff/D71573533/)

[ghstack-poisoned]
